utils/dataset_utils.py

```diff
@@ -69,7 +73,31 @@ def get_dataloaders(args):
     #     # transform to spectrograms, or keep at waveform?
     else:
         raise ValueError("Unknown dataset")
-
+    
+    if args.prev_run_name_for_dynamics:
+        # get subset from dataset using previous run dynamics
+        gold_label_probabilities, confidence, variability, correctness, forgetfulness = get_training_dynamics_from_run_name(args.wandb_project_name, 'l46_datamaps', args.prev_run_name_for_dynamics)
+        selected_indices = get_data_subset(
+            list(range(len(train_dataset))),
+            variability,
+            confidence,
+            correctness,
+            forgetfulness,
+            p_easytolearn=args.p_easytolearn,
+            p_ambiguous=args.p_ambiguous,
+            p_hardtolearn=args.p_hardtolearn,
+            p_variability=args.p_variability,
+            selector_variability=args.selector_variability, 
+            p_confidence=args.p_confidence,
+            selector_confidence=args.selector_confidence,  
+            p_correctness=args.p_correctness,
+            selector_correctness=args.selector_correctness,  
+            p_forgetfulness=args.p_forgetfulness,
+            selector_forgetfulness=args.selector_forgetfulness  
+        )
+
+        # Create a Subset object for the selected indices
+        train_dataset = Subset(train_dataset, selected_indices)
     # Split the training dataset into train and validation
     train_size = int(0.8 * len(train_dataset))
     val_size = len(train_dataset) - train_size
```

Comment:
Although `train_size` will now be 13200, it's not guaranteed to be the true 0.33 of the original 40k.

Also, the validation dataset should not be influenced by the subset selection. Perhaps the train/val split should be done deterministically for each run, and only then select the subset after the split.

Reply:
Agreed, will do